### DISCUSSION TITLE
Raw query only works if I cast the values. Why?

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#17019] [Q&A]
Why does it only work if I cast the variables?
```
  const from = '2022-12-01'
  const to = '2022-12-31'

//   this works
//_____________________
   const query =
     await prisma.$queryRaw`SELECT * FROM "Invoice_number" WHERE "created_at" BETWEEN ${from}::DATE AND ${to}::DATE`

//   this doesn't work
//_____________________
  const query =
    await prisma.$queryRaw`SELECT * FROM "Invoice_number" WHERE "created_at" BETWEEN ${from} AND ${to}`
  
res.status(200).json({ facturas: query })
```

**Comment by pantharshit00** [comment]
Hello @gmcamposano,

`queryRaw` internally converts any string interpolation that you are doing to a prepared statement variable to prevent SQL injection. So in this case, that variable is going as a string to the driver so that is why it is throwing an error. 

You can open a [new issue](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind/bug&template=bug_report.yml) if you want to see a change in this behaviour.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
